.github/workflows-source/ci-main.yml

```diff
@@ -112,17 +112,27 @@ jobs:
         shell: bash
         run: |
           set -xeuo pipefail
-
           # Determine which tests to skip and append 'long_test' for pull requests or merge groups
-          EXCLUDED_TEST_TAGS=(system_test_hourly system_test_nightly system_test_nightly_nns system_test_staging system_test_hotfix system_test_benchmark fuzz_test fi_tests_nightly nns_tests_nightly)
-          [[ "${{ github.event_name }}" =~ ^(pull_request|merge_group)$ ]] && EXCLUDED_TEST_TAGS+=(long_test)
-
+          EXCLUDED_TEST_TAGS=(
+              system_test_hourly
+              system_test_nightly
+              system_test_nightly_nns
+              system_test_staging
+              system_test_hotfix
+              system_test_benchmark
+              fuzz_test
+              fi_tests_nightly
+              nns_tests_nightly
+          )
+          if [[ "${{ github.event_name }}" =~ ^(pull_request|merge_group)$ ]]; then
+              if [[ "${GITHUB_EVENT_NAME}" == "merge_group" || "${RUN_ON_DIFF_ONLY:-}" == "true" ]]; then
```

Comment:
Why mix GitHub syntax `${{ github.event_name }}` with env variables meaning the same thing `${GITHUB_EVENT_NAME}` ?

BTW this would be a shorter condition without requiting nested ifs:
```
if [[ "${GITHUB_EVENT_NAME}" == "merge_group" || "${GITHUB_EVENT_NAME}" == "pull_request" && "${RUN_ON_DIFF_ONLY:-}" == "true" ]]; then
```
```

Reply:
It's actually a mistake. Should be `${{ github.event_name }}`.

Reply:
I've updated it. Left it as it was as I think it's more easy to understand.